Users

Synkatathesis opened the pull request:
added bcryptjs, created a model for users and requests to create update and delete users, still a work in progress but give me your thoughts on this

seanjhardy reviewed (changes requested):
Looking awesome :D One thing I'd say is that the "badges" for a user, are equivalent to their completed skills. We can fetch the badge by their skill instead of storing it separately. Also we need completed skills and current skills. The skills/items should reference ObjectIDs to other objects in the DB. Also the default icon path doesn't work for me (also not sure if we'll even store it? but probably fine to leave it in for now)

Synkatathesis commented:
> Looking awesome :D One thing I'd say is that the "badges" for a user, are equivalent to their completed skills. We can fetch the badge by their skill instead of storing it separately. Also we need completed skills and current skills. The skills/items should reference ObjectIDs to other objects in the DB. Also the default icon path doesn't work for me (also not sure if we'll even store it? but probably fine to leave it in for now)

Thanks ! I went and fixed a few things, yeah the icon path changed since the last time I used it so I changed it, should be fine now

seanjhardy commented:
Yeah, but we'll also need backwards compatibility with users who don't have discord and who want to use the app. I think what we need to do is store the username/password, as well as any other integration/signin methods like discord, so both users can authenticate with whatever service they choose. Might deserve more of a discussion than github comments imo

swatarianess commented on src/models/user.model.js:
```diff
@@ -0,0 +1,69 @@
+        type:String,
+        required:true,
+        default:
+            "https://icon-library.com/icon/anonymous-avatar-icon-25.html"
```
Replace this with <https://icon-library.com/images/anonymous-avatar-icon/anonymous-avatar-icon-25.jpg> as its a direct link to the image file. 👍🏾

Synkatathesis commented:
There all fixed for now, feel free to tell me if you think there's anything else wrong or missing
